src/react/components/CardMarkdownContent.tsx

```diff
@@ -0,0 +1,58 @@
+import { TFile } from "obsidian";
+import { Fragment } from "preact";
+import { useState, useRef, useEffect } from "preact/hooks";
+
+import CardMarkdownContentRenderer from "~/react/components/CardMarkdownContentRenderer";
+import { useAppMount } from "~/react/context/app-mount-provider";
+import { useIntersectionObserver } from "~/react/utils/intersection-observer";
+
+interface CardMarkdownContentProps {
+  file: TFile;
+}
+
+export default function CardMarkdownContent(props: CardMarkdownContentProps) {
+  const { app } = useAppMount();
+  const { file } = props;
+  const { vault } = app;
+  const [content, setContent] = useState("");
+
+  const ref = useRef<HTMLDivElement | null>(null);
+  const entry = useIntersectionObserver(ref, {
+    rootMargin: "33%",
+    freezeOnceVisible: true,
+  });
+  const isVisible = !!entry?.isIntersecting;
+  const linesToKeep = 30;
+
+  useEffect(() => {
+    (async () => {
+      if (isVisible) {
+        let c = await vault.cachedRead(file);
+
+        // the idea of the below is to trim the content to the first n lines
+        let meta = "";
+        if (c.startsWith("---")) {
+          const i = c.indexOf("---", 3); // second instance
+          meta = c.slice(0, i + 3).trim();
+          c = c.slice(i + 3, c.length).trim();
+        }
+        c = c.split("\n").splice(0, linesToKeep).join("\n");
+        c = [meta, c].join("\n").trim();
+
+        setContent(c);
```

Comment:
[♻️](https://github.com/erikthedeveloper/code-review-emoji-guide#:~:text=:recycle:) Provide better names to `c` and `i` to make it a bit more readable